src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/BlobListenerFactory.cs

```diff
@@ -159,13 +167,19 @@ internal class BlobListenerFactory : IListenerFactory
             IStorageBlobClient blobClient,
             IStorageQueue hostBlobTriggerQueue,
             IMessageEnqueuedWatcher messageEnqueuedWatcher,
+            SingletonManager singletonManager,
             CancellationToken cancellationToken)
         {
             BlobTriggerExecutor triggerExecutor = new BlobTriggerExecutor(hostId, _functionId, _input,
                 BlobETagReader.Instance, new BlobReceiptManager(blobClient),
                 new BlobTriggerQueueWriter(hostBlobTriggerQueue, messageEnqueuedWatcher));
+
             await sharedBlobListener.RegisterAsync(_container, triggerExecutor, cancellationToken);
-            return new BlobListener(sharedBlobListener);
+
+            // we only want the blob discovery/scan listener to be running on a single instance,
+            // so we make it a singleton
+            var blobListener = new BlobListener(sharedBlobListener);
+            return singletonManager.CreateHostSingletonListener(blobListener, "Blobs");
```

Comment:
This will result in a host level lock blob of the form **4496ba9d19da4040aaf65fdf92690e8a/Blob.Listener**